sites/all/modules/unl_multisite/cron.php

```diff
@@ -297,19 +298,109 @@ function unl_add_site($site_path, $uri, $clean_url, $db_prefix, $site_id) {
+function unl_get_site_record($site_id) {
+  $query = db_query('SELECT * FROM {unl_sites} WHERE site_id='.(int)$site_id);
 
+  return $query->fetchAssoc();
+}
+
+/**
+ * @param $from_id
+ * @param $to_id
+ *
+ * @throws Exception
+ */
+function unl_clone_site($from_id, $to_id) {
+  //Get the sites to clone
+  if (!$clone_from_site = unl_get_site_record($from_id)) {
+    //Unable to get the site to clone from
+    throw new Exception('Error while cloning site. Site id ' . $from_id . ' could not be found');
+  }
+
+  if (!$clone_to_site = unl_get_site_record($to_id)) {
+    //Unable to get the site to clone to
+    throw new Exception('Error while cloning site. Site id ' . $to_id . ' could not be found');
+  }
+  
+  //Delete the to site
+  unl_drop_site_tables($clone_to_site['db_prefix']);
+  
+  //Copy tables
   $database = $GLOBALS['databases']['default']['default'];
-  $db_prefix .= '_' . $database['prefix'];
+  $from_db_prefix = $clone_from_site['db_prefix'] .'_' . $database['prefix'];
+  $to_db_prefix = $clone_to_site['db_prefix'] .'_' . $database['prefix'];
+  
+  // Grab the list of tables we need to clone.
+  $tables = getTableNamesWithPrefix($from_db_prefix);
+
+  // Clone the site's tables
+  foreach ($tables as $table) {
+    $from_table = $table;
+    $to_table = str_replace_once($from_db_prefix, $to_db_prefix, $table);
+    try {
+      db_query("CREATE TABLE $to_table LIKE $from_table;");
+      db_query("INSERT $to_table SELECT * FROM $from_table;");
+    } catch (PDOException $e) {
+      // probably already there?
+    }
+  }
+
+  //Copy over files
+  $from_path = escapeshellarg(DRUPAL_ROOT . '/sites/' . unl_get_sites_subdir($clone_from_site['uri']) . '/files/');
+  $to_path = escapeshellarg(DRUPAL_ROOT . '/sites/' . unl_get_sites_subdir($clone_to_site['uri']) . '/files/');
+
+  //Note: .htaccess is read only and can not be replaced
+  $command = "cp -r $from_path $to_path";
+
+  exec($command, $output, $status);
+
+  if (0 !== $status) {
+    echo 'Warning while cloning site. command failed: ' . $command . PHP_EOL;
+  }
+  
+  //Clean up with drush
+  $uri = escapeshellarg($clone_to_site['uri']);
+  
+  //Remove the primary base url
+  $command = DRUPAL_ROOT."/sites/all/modules/drush/drush -y --uri=$uri vdel unl_primary_base_url 2>&1";
+  exec($command, $output, $status);
+  
+  if (0 !== $status) {
+    echo 'Warning while cloning site. command failed: ' . $command . PHP_EOL;
+  }
+  
+  //Clear the cache
+  $command = DRUPAL_ROOT."/sites/all/modules/drush/drush -y --uri=$uri cc all 2>&1";
+  exec($command, $output, $status);
+
+  if (0 !== $status) {
+    echo 'Warning while cloning site. command failed: ' . $command . PHP_EOL;
+  }
+}
+
+function getTableNamesWithPrefix($prefix) {
+  $database = $GLOBALS['databases']['default']['default'];
+
+  $query = "";
+  $query .= "SELECT `table_name`";
+  $query .= " FROM INFORMATION_SCHEMA.TABLES";
+  $query .= " WHERE `table_schema` = '".db_escape_table($database['database'])."'";
```

Comment:
Just to be super cautious, I left it in there but am now escaping it The Right Way™